examples/daal/cpp/source/k_nearest_neighbors/bf_knn_dense_search_batch.cpp

```diff
@@ -0,0 +1,194 @@
+  NumericTablePtr testGroundTruth =
+      HomogenNumericTable<>::create(1, 0, NumericTable::doNotAllocate);
+  NumericTablePtr mergedData(new MergedNumericTable(testData, testGroundTruth));
+
+  /* Retrieve the data from input file */
+  testDataSource.loadDataBlock(mergedData.get());
+
+  /* Create algorithm objects for BF kNN prediction with the default method */
+  bf_knn_classification::prediction::Batch<> algorithm;
+
+  /* Pass the testing data set and trained model to the algorithm */
+  algorithm.input.set(classifier::prediction::data, testData);
+  algorithm.input.set(classifier::prediction::model,
+                      trainingResult->get(classifier::training::model));
+  algorithm.parameter().k = kNeighbors;
+  algorithm.parameter().nClasses = nClasses;
+  algorithm.parameter().resultsToCompute =
+      bf_knn_classification::computeDistances |
+      bf_knn_classification::computeIndicesOfNeighbors;
+
+  /* Compute prediction results */
+  algorithm.compute();
+
+  /* Retrieve algorithm results */
+  predictionResult = algorithm.getResult();
+}
+
+void readGroundTruth(NumericTablePtr &groundTruthIndices,
+                     NumericTablePtr &groundTruthDistances) {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the test data from
+   * a .csv file */
+  FileDataSource<CSVFeatureManager> indicesDataSource(
+      groundTruthIndicesFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+  FileDataSource<CSVFeatureManager> distancesDataSource(
+      groundTruthDistancesFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for testing data and labels */
+  groundTruthIndices =
+      HomogenNumericTable<>::create(kNeighbors, 0, NumericTable::doNotAllocate);
+  groundTruthDistances =
+      HomogenNumericTable<>::create(kNeighbors, 0, NumericTable::doNotAllocate);
+
+  /* Retrieve the data from input file */
+  indicesDataSource.loadDataBlock(groundTruthIndices.get());
+  distancesDataSource.loadDataBlock(groundTruthDistances.get());
+}
+
+void printIndicesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult) {
+  NumericTablePtr indicesResult =
+      predictionResult->get(bf_knn_classification::prediction::indices);
+  printNumericTable(testGroundTruth,
+                    "Indices Ground Truth (first 10 rows):", 10);
+  printNumericTable(indicesResult, "Computed Indices (first 10 rows):", 10);
+}
+
+void printDistancesResults(
+    NumericTablePtr &testGroundTruth,
+    bf_knn_classification::prediction::ResultPtr &predictionResult) {
+  NumericTablePtr distancesResult =
+      predictionResult->get(bf_knn_classification::prediction::distances);
+  printNumericTable(testGroundTruth,
+                    "Distances Ground Truth (first 10 rows):", 10);
+  printNumericTable(distancesResult, "Computed Distances (first 10 rows):", 10);
+}
```

Comment:
Can printNumericTables be used here instead?

Reply:
Probably no since now `printNumericTables` is hardcoded to set precision equals 3 for the first table and precision equals 0 for the second one. It make it almost unusable for printing 2 floating-point typed tables.